File: 234-palindrome-linked-list/palindrome-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        stack<int> container;
        ListNode* temp;

        temp = head;

        while (temp != nullptr) {
            container.push(temp->val);
            temp = temp->next;
        }
    
        
        while (head != nullptr) {
            if (head->val != container.top()){
                return false;
            }

            container.pop();
            head = head->next;
        }
        return true;
    }
};

File: 234-palindrome-linked-list/palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        stack =[]

        temp = head

        while temp != None:
            stack.append(temp.val)
            temp = temp.next


        while head != None:
            if head.val != stack.pop():
                return False
            head = head.next
        
        return True

        